tools/.golangci.yaml

```diff
@@ -19,14 +19,74 @@ linters:
     # - varcheck
     - goimports
     - ineffassign
-    # - revive     # TODO: enable by #16610
+    - revive
     - staticcheck
     # - stylecheck # TODO: enable by #16610
     # - unused     # TODO: enable by #16610
     # - unconvert  # Remove unnecessary type conversions TODO: enable by #16610
 linters-settings: # please keep this alphabetized
   goimports:
     local-prefixes: go.etcd.io # Put imports beginning with prefix after 3rd-party packages.
+  revive:
+    ignore-generated-header: false
+    severity: error
+    confidence: 0.8
+    enable-all-rules: false
+    rules:
+      - name: blank-imports
+        severity: error
+        disabled: false
+      - name: context-as-argument
+        severity: error
+        disabled: false
+      - name: dot-imports
+        severity: error
+        disabled: false
+      - name: error-return
+        severity: error
+        disabled: false
+      - name: error-naming
+        severity: error
+        disabled: false
+      - name: if-return
+        severity: error
+        disabled: false
+      - name: increment-decrement
+        severity: error
+        disabled: false
+      - name: var-declaration
+        severity: error
+        disabled: false
+      - name: package-comments
+        severity: error
+        disabled: false
+      - name: range
+        severity: error
+        disabled: false
+      - name: receiver-naming
+        severity: error
+        disabled: false
+      - name: time-naming
+        severity: error
+        disabled: false
+      - name: indent-error-flow
+        severity: error
+        disabled: false
+      - name: errorf
+        severity: error
+        disabled: false
+      - name: context-keys-type
+        severity: error
+        disabled: false
+      # TODO: enable the following rules
```

Comment:
What's the plan for the TODO?

Reply:
Flip `disabled: true`?

Reply:
I mean why they are disabled now and when to enable them?

It isn't a big deal though.

Reply:
TODO in tests/revive.toml

Reply:
@ahrtr The TODO was defined in the tests/revive.toml. I can file issue to track this~